internal/landing/landing.go

```diff
@@ -8,22 +8,74 @@ package landing
 
 import (
 	"embed"
+	"fmt"
+	"html/template"
 	"io/fs"
 	"net/http"
+	"path/filepath"
+
+	"github.com/talos-systems/discovery-service/internal/state"
 )
 
 //go:embed "html/index.html"
 var static embed.FS
 
+type ClusterData struct {
+	ClusterId  string
+	Affiliates []*state.AffiliateExport
+}
+
+func InspectHandler(w http.ResponseWriter, req *http.Request, state *state.State) {
+
+	clusterData := ClusterData{}
+
+	if req.Method == "POST" {
+		if err := req.ParseForm(); err != nil {
+			panic(err)
+		}
+		if formData := req.PostForm["cluster-id"]; formData != nil {
+			clusterData.ClusterId = formData[0]
+		}
+		fetchedCluster := state.GetCluster(clusterData.ClusterId)
+
+		clusterData.Affiliates = fetchedCluster.List()
+	}
+
+	file, err := filepath.Abs("internal/landing/html/inspect.html")
+	if err != nil {
+		fmt.Println("Abs: file not found!!", err)
+	}
+
+	inspectPage, err := template.ParseFiles(file)
+	if err != nil {
+		fmt.Println("ParseFiles: file not found!!", err)
+	}
+
+	err = inspectPage.Execute(w, clusterData)
+	if err != nil {
+		fmt.Println("Error executing template :", err)
```

Comment:
more or less same thing as above about error handling - one way is to extract this out and have a common wrapper which inspects error and prints back error message if required